src/game/Tactical/PathAI.cc

```diff
@@ -2529,6 +2648,54 @@ UINT8 InternalDoorTravelCost(const SOLDIERTYPE* pSoldier, INT32 iGridNo, UINT8 u
 
 		switch( ubMovementCost )
 		{
+			case TRAVELCOST_DOORS_CLOSED_W_SW:
+				fDoorIsObstacleIfClosed = TRUE;
+				iDoorGridNo = iGridNo + DirIncrementer[WEST];
+				iDoorGridNo2 = iGridNo + DirIncrementer[SOUTHWEST];
+				ubReplacementCost = TRAVELCOST_DOOR;
+				break;
```

Comment:
This is quite repetitive. Can the whole switch be moved to a separate function that then this and FindBestPath could use (and we get to maintain only one copy)?